lib-image/src/main/java/ai/kognition/pilecv4j/image/Utils.java

```diff
@@ -1137,6 +1137,26 @@ public static Size preserveAspectRatio(final Size originalMatSize, final Size ne
             : new Size(Math.round(originalMatSize.width * scale), Math.round(originalMatSize.height * scale));
     }
 
+    public static Size preserveAspectRatio(final Mat mat, final Size newSize) {
+        return preserveAspectRatio(mat.size(), newSize);
+    }
+
+    /**
+     *
+     * @param originalMatSize The size of the original image
+     * @param newSize The desired size of the new image
+     * @return The size of the new image, which matches the height or width of {@param newSize} such that the image does not exceed those dimensions while
+     * preserving the size.
+     */
+    public static Size preserveAspectRatio(final Size originalMatSize, final Size newSize) {
```

Comment:
Change the name of this function- otherwise, your new function will be called where the old one was- that'll make refactoring it harder. Rename to maybe "scaleWhilePreservingAspectRatio" and rename `newSize` to `maxSize`?

Reply:
Done